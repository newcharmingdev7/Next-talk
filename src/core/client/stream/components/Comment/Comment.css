.root {
}
.topBar {
  margin-bottom: calc(0.5 * var(--spacing-unit));
}
.footer:not(:empty) {
  margin-top: var(--spacing-unit);
}
.reply {
  margin-top: var(--spacing-unit);
}
