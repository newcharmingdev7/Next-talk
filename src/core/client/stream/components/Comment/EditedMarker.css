.root {
  composes: detail from "talk-ui/shared/typography.css";
}
