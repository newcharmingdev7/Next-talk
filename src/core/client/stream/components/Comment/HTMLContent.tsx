import dompurify from "dompurify";
import React, { StatelessComponent } from "react";

import styles from "./HTMLContent.css";

interface HTMLContentProps {
  children: string;
}

const HTMLContent: StatelessComponent<HTMLContentProps> = ({ children }) => (
  <div
    className={styles.root}
    dangerouslySetInnerHTML={{ __html: dompurify.sanitize(children) }}
  />
);

export default HTMLContent;
