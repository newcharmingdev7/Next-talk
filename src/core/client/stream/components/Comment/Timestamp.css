.root {
  composes: timestamp from "talk-ui/shared/typography.css";
}
