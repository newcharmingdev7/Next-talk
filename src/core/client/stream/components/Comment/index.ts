export { default, default as IndentedComment } from "./IndentedComment";
