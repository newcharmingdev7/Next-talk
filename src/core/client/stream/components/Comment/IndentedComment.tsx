import React, { StatelessComponent } from "react";

import { PropTypesOf } from "talk-framework/types";

import Indent from "../Indent";
import Comment from "./Comment";

export interface IndentedCommentProps extends PropTypesOf<typeof Comment> {
  indentLevel?: number;
}

const IndentedComment: StatelessComponent<IndentedCommentProps> = props => {
  const { indentLevel, ...rest } = props;
  const CommentElement = <Comment {...rest} />;
  const CommentwithIndent =
    (indentLevel && <Indent level={indentLevel}>{CommentElement}</Indent>) ||
    CommentElement;
  return CommentwithIndent;
};

export default IndentedComment;
