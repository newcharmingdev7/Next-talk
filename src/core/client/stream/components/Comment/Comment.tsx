import React, { ReactElement, StatelessComponent } from "react";

import { Flex } from "talk-ui/components";

import * as styles from "./Comment.css";
import HTMLContent from "./HTMLContent";
import Timestamp from "./Timestamp";
import TopBarLeft from "./TopBarLeft";
import Username from "./Username";

export interface CommentProps {
  className?: string;
  author: {
    username: string | null;
  } | null;
  body: string | null;
  createdAt: string;
  topBarRight?: React.ReactNode;
  footer?: React.ReactNode;
}

const Comment: StatelessComponent<CommentProps> = props => {
  return (
    <div role="article" className={styles.root}>
      <Flex
        className={styles.topBar}
        direction="row"
        justifyContent="space-between"
      >
        <TopBarLeft>
          {props.author &&
            props.author.username && (
              <Username>{props.author.username}</Username>
            )}
          <Timestamp>{props.createdAt}</Timestamp>
        </TopBarLeft>
        {props.topBarRight && <div>{props.topBarRight}</div>}
      </Flex>
      <HTMLContent>{props.body || ""}</HTMLContent>
      <Flex className={styles.footer} direction="row" itemGutter="half">
        {props.footer}
      </Flex>
    </div>
  );
};

export default Comment;
