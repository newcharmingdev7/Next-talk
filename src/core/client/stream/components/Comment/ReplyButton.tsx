import { Localized } from "fluent-react/compat";
import React, { EventHandler, MouseEvent, StatelessComponent } from "react";

import { Button, ButtonIcon, MatchMedia } from "talk-ui/components";

interface Props {
  onClick?: EventHandler<MouseEvent<HTMLButtonElement>>;
  active?: boolean;
}

const ReplyButton: StatelessComponent<Props> = props => (
  <Button
    onClick={props.onClick}
    variant="ghost"
    size="small"
    active={props.active}
  >
    <MatchMedia gtWidth="xs">
      <ButtonIcon>reply</ButtonIcon>
    </MatchMedia>
    <Localized id="comments-replyButton-reply">
      <span>Reply</span>
    </Localized>
  </Button>
);

export default ReplyButton;
