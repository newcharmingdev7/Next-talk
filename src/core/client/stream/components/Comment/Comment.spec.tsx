import { shallow } from "enzyme";
import React from "react";

import { PropTypesOf } from "talk-framework/types";

import Comment from "./Comment";

it("renders username and body", () => {
<<<<<<< HEAD:src/core/client/stream/components/Comment.spec.tsx
  const props = {
    id: "id",
    author: {
      username: "Marvin",
    },
    body: "Woof",
  };
  const wrapper = shallow(<Comment {...props} />);
  expect(wrapper).toMatchSnapshot();
});

it("renders with gutterBottom", () => {
  const props = {
    id: "id",
=======
  const props: PropTypesOf<typeof Comment> = {
>>>>>>> abfa39529ff398007bb67cd814c4217a97bdb38b:src/core/client/stream/components/Comment/Comment.spec.tsx
    author: {
      username: "Marvin",
    },
    body: "Woof",
    createdAt: "1995-12-17T03:24:00.000Z",
  };
  const wrapper = shallow(<Comment {...props} />);
  expect(wrapper).toMatchSnapshot();
});
