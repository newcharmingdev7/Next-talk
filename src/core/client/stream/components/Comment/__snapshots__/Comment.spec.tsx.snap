// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders username and body 1`] = `
<div
  className="Comment-root"
  role="article"
>
  <withPropsOnChange(Flex)
    className="Comment-topBar"
    direction="row"
    justifyContent="space-between"
  >
    <TopBarLeft>
      <Username>
        Marvin
      </Username>
      <Timestamp>
        1995-12-17T03:24:00.000Z
      </Timestamp>
    </TopBarLeft>
  </withPropsOnChange(Flex)>
  <HTMLContent>
    Woof
  </HTMLContent>
  <withPropsOnChange(Flex)
    className="Comment-footer"
    direction="row"
    itemGutter="half"
  />
</div>
`;
